src/DotFlyer.Api.Tests/APITests.cs

```diff
@@ -411,6 +411,176 @@ public async Task Post_Email_ShouldBe_Idempotent_When_Same_MessageId_Is_Reused()
+
+        // Wait for the email to be processed and ingested into ADX
+        EmailData emailData = await _cslQueryProvider!
+            .WaitSingleQueryResult<EmailData>(
+                $"[\"{EmailTable.Instance.TableName}\"] | where Subject == \"{emailMessage.Subject}\"",
+                TimeSpan.FromMinutes(10),
+                _cancellationToken);
+
+        emailData.Should().NotBeNull();
+        emailData.Subject.Should().Be(emailMessage.Subject);
+        emailData.FromEmail.Should().Be(emailMessage.From.Email);
+        emailData.FromName.Should().Be(emailMessage.From.Name);
+        emailData.SendGridStatusCodeInt.Should().Be(202);
+        emailData.SendGridStatusCodeString.Should().Be("Accepted");
+    }
+
+    [TestMethod]
+    public async Task Post_Email_WithSalesReportTemplate_ShouldReturn_200_And_RenderTemplate()
+    {
+        var testGuid = Guid.NewGuid();
+        
+        var emailMessage = new EmailMessage
+        {
+            Subject = $"Sales Report Serialization Test - {testGuid}",
+            Body = "Fallback body if template fails",
+            From = new Contact
+            {
+                Email = _senderEmail,
+                Name = "Integration Test"
+            },
+            To =
+            [
+                new Contact
+                {
+                    Email = _receiverEmail,
+                    Name = "Integration Test Destination"
+                }
+            ],
+            TemplateId = EmailTemplateIds.SalesReport,
+            TemplateModel = new SalesReportModel
+            {
+                Title = $"Serialization Test Report - {testGuid}",
+                ClientName = "Serialization Test Corp",
+                ContactEmailAddress = "serialization@test.com"
+            },
+            Tags = new Dictionary<string, string>
+            {
+                { "TestName", "Email Template Serialization Test" }
+            }
+        };
+
+        // Call API to verify serialization and deserialization
+        var httpClient = GetHttpClient(await GetSenderAccessTokenAsync());
+        var response = await httpClient.PostAsync("dotflyer/email", GetStringContent(emailMessage), _cancellationToken);
+
+        var responseContent = await response.Content.ReadAsStringAsync(_cancellationToken);
+        response.StatusCode.Should().Be(HttpStatusCode.OK, $"Response content: {responseContent}");
+
+        // Wait for the email to be processed and ingested into ADX
+        EmailData emailData = await _cslQueryProvider!
+            .WaitSingleQueryResult<EmailData>(
+                $"[\"{EmailTable.Instance.TableName}\"] | where Subject == \"{emailMessage.Subject}\"",
+                TimeSpan.FromMinutes(10),
+                _cancellationToken);
+
+        emailData.Should().NotBeNull();
+        emailData.Subject.Should().Be(emailMessage.Subject);
+        emailData.FromEmail.Should().Be(emailMessage.From.Email);
```

Comment:
I am not a stickler for code copy/paste in tests but here it seems we might benefit?